### PR TITLE
[TASK] Remove outdated IE6 condition in base tag

### PR DESCRIPTION
**Pull request by ayacoo**
Related #10502

### CODE REVIEW

**Comment by GuySartorelli** [comment]
There are some failing tests in the CI - please resolve those.

**Comment by ayacoo** [comment]
From my point of view, the first two assertions (line 958 - 978) can be removed completely. However, as I am new here, I did not want to simply decide this and therefore adapted the tests simply for the time being.

**Comment by GuySartorelli** [comment]
One of the tests is still failing.
I agree that it makes sense to remove the tests that were previously testing the functionality which you are removing.

**Comment by ayacoo** [comment]
On the last run-through, I am puzzled as to why no slash is added there. Does anyone have any ideas about this? Apparently a placeholder %base_tag% in the test is replaced, but looking around the project, I don't have a clever idea where this happens.

File: src/View/SSViewer.php [hunk 1]
```diff
@@ -851,13 +851,6 @@ public function setTemplateFile($type, $file)
...
     {
-        $base = Director::absoluteBaseURL();
-
-        // Is the document XHTML?
-        if (preg_match('/<!DOCTYPE[^>]+xhtml/i', $contentGeneratedSoFar ?? '')) {
-            return "<base href=\"$base\" />";
-        } else {
-            return "<base href=\"$base\"><!--[if lte IE 6]></base><![endif]-->";
-        }
+        return '<base href="' . Director::absoluteBaseURL() . '" />';
```
Comment (GuySartorelli) [inline src/View/SSViewer.php:854]:
```suggestion
        return '<base href="' . Director::absoluteBaseURL() . '">';
```
Modern browsers don't require this to be self-closing. That's true of any browser that supports HTML4+

File: tests/php/View/SSViewerTest.php [hunk 2]
```diff
@@ -955,35 +955,13 @@ public function testBaseTagGeneration()
...
-        );
-
         // Check that the content negotiator converts to the equally legal formats
         $negotiator = new ContentNegotiator();
 
         $response = new HTTPResponse($this->render($tmpl1));
         $negotiator->html($response);
         $this->assertMatchesRegularExpression(
-            '/<head><base href=".*"><!--\[if lte IE 6\]><\/base><!\[endif\]--><\/head>/',
+            '/<head><base href=".*" \/><\/head>/',
```
Comment (GuySartorelli) [inline tests/php/View/SSViewerTest.php:964]:
```suggestion
            '/<head><base href=".*"><\/head>/',
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

**Comment by GuySartorelli** [comment]
Not sure why the slash wasn't being added - but it's actually better that it isn't added since this tag doesn't need to be self-closing.

**Pull request closed without merging**